packages/unuse-reactivity/src/unSignal/index.ts

```diff
@@ -102,10 +90,26 @@ export function unSignal<T>(initialValue?: T): UnSignal<T> {
 
       return value;
     },
-    peek: () => state.value,
-    set: setter,
-    update: (updater) => {
-      setter(updater(state.value));
+    peek() {
+      const prev = setCurrentSub(undefined);
+      const val = this.get();
+      setCurrentSub(prev);
+      return val;
+    },
+    set(newValue) {
+      if (state.value !== (state.value = newValue)) {
+        state.flags = 17 as ReactiveFlags.Mutable | ReactiveFlags.Dirty;
+        const subs = state.subs;
+        if (subs !== undefined) {
+          propagate(subs);
+          if (!REACTIVITY_STATE.batchDepth) {
+            flush();
+          }
+        }
+      }
+    },
+    update(updater) {
+      this.set(updater(state.value));
```

Comment:
Should we also change this line?

```suggestion
      this.set(updater(this.peek()));
```

Reply:
I understand your logic. For some reason it seems to me that the insert operation can be done without it. ))

My motivation was to save the source code to get the value

Reply:
We don't have to uniform them if they don't need to, and we should be consistent in `unSignal` itself.